src/servers/src/http/handler.rs

```diff
@@ -117,17 +123,49 @@ pub async fn sql(
         Ok(outputs) => outputs,
     };
 
-    let resp = match format {
+    let mut resp = match format {
         ResponseFormat::Arrow => ArrowResponse::from_output(outputs).await,
         ResponseFormat::Csv => CsvResponse::from_output(outputs).await,
         ResponseFormat::Table => TableResponse::from_output(outputs).await,
         ResponseFormat::GreptimedbV1 => GreptimedbV1Response::from_output(outputs).await,
         ResponseFormat::InfluxdbV1 => InfluxdbV1Response::from_output(outputs, epoch).await,
     };
 
+    if let Some(RequestSource::Dashboard) = source {
+        resp = from_dashboard(resp, query_ctx).await;
+    }
     resp.with_execution_time(start.elapsed().as_millis() as u64)
 }
 
+pub async fn from_dashboard(resp: HttpResponse, ctx: QueryContextRef) -> HttpResponse {
+    let HttpResponse::GreptimedbV1(response) = resp else {
+        return resp;
+    };
+    let GreptimedbV1Response {
```

Comment:
I think we should add a field of `total_rows: u32` in `GreptimedbV1Response`, to inform the Dashboard there might be some rows "limit"ed.

Reply:
To be precise, it should be added to `HttpRecordsOutput`.  It corresponds to the execution result of each statement.
https://github.com/GreptimeTeam/greptimedb/blob/d0820bb26dfd6962c7a2f92980d90ee25841280c/src/servers/src/http.rs#L190-L192